src/TestUtils/src/Microsoft.Maui.IntegrationTests/Utilities/BuildWarningsUtilities.cs

```diff
@@ -0,0 +1,632 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Logging.StructuredLogger;
+using System.Linq;
+using static Microsoft.Maui.IntegrationTests.TemplateTests;
+
+namespace Microsoft.Maui.IntegrationTests
+{
+	public class WarningsPerFile
+	{
+		public string File { get; set; } = string.Empty;
+		public List<WarningsPerCode> WarningsPerCode { get; set; } = new List<WarningsPerCode>();
+	}
+
+	public class WarningsPerCode
+	{
+		public string Code { get; set; } = string.Empty;
+		public List<string> Messages { get; set; } = new List<string>();
+	}
+
+	public static class BuildWarningsUtilities
+	{
+		public static List<WarningsPerFile> ExpectedNativeAOTWarnings
+		{
+			get => expectedNativeAOTWarnings;
+		}
+
+		// We rely on the fact that expected file paths are stored as relative to the repo root (e.g., src/Core/...).
+		// While the actual file paths are always full paths and can have different repo roots (e.g., building locally or on CI).
+		private static bool CompareWarningsFilePaths(this string actual, string expected) => actual.Contains(expected, StringComparison.Ordinal);
+
+		private static string NormalizeFilePath(string file) => file.Replace("\\\\", "/", StringComparison.Ordinal).Replace("\\", "/", StringComparison.Ordinal);
```

Comment:
Minor speed improvement:

```suggestion
		private static string NormalizeFilePath(string file) => file.Replace("\\\\", "/", StringComparison.Ordinal).Replace('\\', '/');
```

Reply:
Thank you. Corrected.